### PR TITLE
feat(Marker.Drag): autoPan implementation

### LINKED ISSUE #3572: Move map at the edge of map on marker dragging
when you drag a marker to the edge of the map, the map would pan along with you, making it easy to move your marker long distances.  But, as far as I can tell, Leaflet doesn't do this.  In  order to move your marker to a location that isn't currently visible, you have to zoom out, move your marker, and then zoom back in.

Am I missing something?

### PR DESCRIPTION
**Pull request by alenaksu**
Hi,
this is an implementation of auto panning for the marker (should fix #3572). Unlike #3643, this doesn't modify  the Draggable class.
It also implements a sort of incremental speed that will increase as far as you move closer the edge.

Hope it helps!

### CODE REVIEW

**Comment by alenaksu** [comment]
I saw your changes and I have to say that we have decreased the speed in our app too, so I think 10 is better.

I'm looking forward to see this PR merged :)

File: src/layer/marker/Marker.js [hunk 1]
```diff
@@ -36,6 +36,18 @@ export var Marker = Layer.extend({
...
 
+		// @option autoPan: Boolean = false
+		// Set it to `true` if you want the map to do panning animation when marker hits the edges.
+		autoPan: false,
+
+		// @option autoPanPadding: Point = Point(5, 5)
+		// Equivalent of setting both top left and bottom right autopan padding to the same value.
+		autoPanPadding: [50, 50],
+
+		// @option autoPanSpeed: Number = 100
```
Comment (theashyster) [inline src/layer/marker/Marker.js]: Isn't this value responsible about what will be shown in the docs as the default value?
If that is the case then these are wrong for both `autoPanPadding` and `autoPanSpeed`

File: spec/suites/layer/marker/Marker.DragSpec.js [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+				}
+			});
+			var toucher = hand.growFinger('mouse');
+
+			toucher.wait(100).moveTo(300, 280, 0)
+				.down().moveBy(5, 0, 20).moveBy(290, 32, 1000).wait(100).up().wait(100);
+		});
+	});
+
+	// describe("#setIcon", function () {
```
Comment (theashyster) [inline spec/suites/layer/marker/Marker.DragSpec.js]: Why commenting out such a large chunk of tests and not just removing them?

Comment (perliedman) [inline spec/suites/layer/marker/Marker.DragSpec.js]: Ouch, my bad. `Marker.DragSpec.js` is a new file, but as you can see I copied it from another file and forgot to remove the old contents.

Thanks for keeping an eye on me! 😄

File: src/layer/marker/Marker.js (hunk 1 above)
Comment (perliedman) [inline src/layer/marker/Marker.js]: :+1: Good catch!

File: spec/suites/layer/marker/Marker.DragSpec.js (hunk 2 above)
Comment (theashyster) [inline spec/suites/layer/marker/Marker.DragSpec.js]: Heh, yep, saw that. No problem 😄

**Pull request merged**